logic/src/commonMain/kotlin/com/wire/kalium/logic/feature/client/RegisterClientUseCase.kt

```diff
@@ -0,0 +1,28 @@
+package com.wire.kalium.logic.feature.client
+
+import com.wire.kalium.logic.data.client.ClientRepository
+import com.wire.kalium.logic.data.client.RegisterClientParam
+import com.wire.kalium.logic.failure.AuthenticationFailure
+import com.wire.kalium.logic.functional.Either
+import kotlinx.coroutines.flow.Flow
+import kotlinx.coroutines.flow.flow
+
+
+class RegisterClientUseCase(
+    private val clientRepository: ClientRepository
+) {
+    suspend operator fun invoke(param: RegisterClientParam): Flow<RegisterClientResult> = flow {
```

Comment:
For now, I think this is good.

But I understand that if I want to use `coreLib` in an app, I should not need to worry about things like `preKey`, `lastPreKey`, etc.

One thing we need to always keep in mind, is what `core` is responsible for, and how one should use it.

This UseCase + Repository combo doens't filter out any complexity or implementation details, it's just a proxy to the network layer.

The current utilization is something like:

```kotlin
coreLogic.userSession(currentUserSession){
    val prekeys = ??
    val lastKey = ??
    val deviceType = ??
    val type = ??
    val capabilities = ??
    val model = ??
    val registerParams = RegisterClientParam(password, preKeys, lastKey, deviceType, type, label, capabilities, model)
    client.registerClient(registerParams)
}
```

I'd expect the CoreLib to be able to get device info by itself on multiple platforms (easily achieved by `expect/actual`), use the `cryptography` module to create prekeys, and only needing needing things that are actual specific to the app/UI. So the consumer would do something like this:

```kotlin
coreLogic.userSession(currentUserSession){
    val registerParams = RegisterClientParam(password, type, clientName)
    client.registerClient(registerParams)
}
```


So, again: For now, I think this is good. This unblocks the sample app. 
But as the core lib gets thicker and implements other repositories and usecases, we can start to make it smarter.

Reply:
i added the `ClientConfig` class for client type, model and so on